File: Assignment44/program2.c
//////////////////////////////////////////////////////////////////////
// File name :       program2.c
// Discription :     It is program which accept matrix and one number 
//                   from user and return frequency of that number.
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

//////////////////////////////////////////////////////////////////////
// Function Name :   CountFrequency
// Discription :     It is used to return frequency of given numbers.
// Input  :          int,int, int** 
// Output :          int
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////

int CountFrequency(int rows, int cols,int **Arr,int iNo)
{
    int iFreq = 0;

    for(int i = 0 ; i < rows ; i++)
    {
        for(int j = 0; j < cols; j++)
        {
            if(Arr[i][j] == iNo)
            {
                iFreq++;
            }
        }
    }
    return iFreq;
} 

int main()
{
    int rows = 0,cols = 0,i = 0,j = 0;
    int iRet = 0,iValue = 0;

    printf("Enter the rows : \n");
    scanf("%d",&rows);

    printf("Enter the cols : \n");
    scanf("%d",&cols);
    
    int **Arr = malloc(rows * sizeof(int *));
    
    for(i = 0; i < rows; i++)
    {
        Arr[i] = malloc(cols * sizeof(int));
    }
    printf("Enter elements:\n");
    for(i = 0; i < rows; i++) {
        for(j = 0; j < cols; j++) {
            scanf("%d", &Arr[i][j]);
        }
    }
    printf("Elements are in Array : \n");

    for(i = 0; i < rows; i++)
    {
        for(j = 0; j < cols; j++)
        {
            printf("%d\t",Arr[i][j]);
        }
        printf("\n");
    }
    printf("\n");

    printf("Enter the number to find occurance of it : \n");
    scanf("%d",&iValue);

    iRet = CountFrequency(rows,cols,Arr,iValue);
    
    printf("Frequency of %d is :%d\n",iValue,iRet);
  
    for(i = 0; i < rows; i++)
    {
        free(Arr[i]);
    }
    free(Arr);
    return 0;
}

File: Assignment46/program2.java
import java.util.Scanner;
//////////////////////////////////////////////////////////////////////
// File name :       program2.java
// Discription :     It is a program to print  pattern.
// Author :          Mangesh Ashok Bedre
// Date :            25/11/2025
//
//////////////////////////////////////////////////////////////////////
class Pattern
{
    
    //////////////////////////////////////////////////////////////////////
    // Function Name :   Display
    // Discription :     It is used to print pattern.
    // Input  :          int ,int
    // Output :          A B C D
    //                   a b c d
    //                   A B C D
    //                   a b c d
    // Author :          Mangesh Ashok Bedre
    // Date :            25/11/2025
    //
    //////////////////////////////////////////////////////////////////////
    public void Display(int rows, int cols)
    {
        int i = 0;
        int j = 0;
        char ch1 = '\0',ch2 = '\0';

        for(i = 0; i < rows; i++)
        {
            for(j = 0,ch1 = 'A',ch2 = 'a'; j < cols; j++,ch1++,ch2++)
            {
                if((i % 2) != 0)
                {
                    System.out.print(ch2+"\t");
                }
                else
                {
                    System.out.print(ch1+"\t");
                }
            }
            System.out.println();
        }
    }
}

class program2
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the rows : ");
        int rows = sobj.nextInt();

        System.out.println("Enter the cols : ");
        int cols = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(rows,cols);
    }
}